CMake configure regression when targeting Xcode/iOS

### Bug Description

The most recent `main` branch can no longer be configured for Xcode (i.e. with `cmake -G Xcode ...`):

```
CMake Error in res/shaders/CMakeLists.txt:
  The custom command generating

    /Users/<user>/git/m1xxx-ios/mixxx/build/res/shaders/.rcc/qrc_waveform_shaders.cpp

  is attached to multiple targets:

    mixxx-lib_waveform_shaders
    mixxx-lib_resources_1

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Version

https://github.com/fwcd/mixxx/commit/218517904ab03d776ecd0fb8cb7bbe3dd5b87850 (based on c368119262913411f94766a3e59735678bd72832)

### OS

macOS 14.5

--- a/res/shaders/CMakeLists.txt
+++ b/res/shaders/CMakeLists.txt
@@ -3,3 +3,10 @@ qt_add_shaders(mixxx-lib "waveform_shaders"
   FILES
       "rgbsignal_qml.frag"
 )
+
+# Workaround for https://bugreports.qt.io/browse/QTBUG-118500 that can be
+# removed once Qt fixes the shader target generation for Xcode.
+# See also https://github.com/mixxxdj/mixxx/issues/13378
+if(CMAKE_GENERATOR STREQUAL "Xcode")
+  add_dependencies(mixxx-lib_resources_1 mixxx-lib_waveform_shaders)
+endif()

